modules/sw_services/device_control/host/device_access_spi_rpi.c

```diff
@@ -13,15 +13,29 @@
 #define DBG(x)
 #define PRINT_ERROR(...)   fprintf(stderr, "Error  : " __VA_ARGS__)
 
+// Number of usec to delay between spi transactions
+static long intertransaction_delay;
+
+// Sleep for intertransaction_delay nanoseconds. Yields to the OS so expect minimum delay to be hundreds
+// of microseconds at least.
+static void apply_intertransaction_delay() {
+	if(0 < intertransaction_delay) {
+		struct timespec req = {0, intertransaction_delay};
```

Comment:
I don't think we need to provide nanosecond resolution worth of delay. The function call overhead itself is hundreds of usecs. Wouldn't something like usleep() will be easier so you don't need to initialise the timespec?

This function would also make sense either being inline or turned into a macro to avoid function call overhead.

Reply:
> this function obsolete; use [nanosleep](https://linux.die.net/man/2/nanosleep)(2) instead.

from the [man page of sleep](https://linux.die.net/man/3/usleep)

The reason usleep takes so long is not function call overhead, but the cost of yielding the thread to the OS and waiting to be scheduled again

Reply:
Okay, if nanosleep is recommended over usleep, lets use nanosleep itself